src/reducers/status/index.js

```diff
@@ -0,0 +1,117 @@
+import { handleActions } from 'redux-actions'
+import reduceReducers from 'reduce-reducers'
+
+import {
+    clearLocalAlert,
+    clearGlobalAlert,
+    setMainLoader
+} from '../../actions/status'
+
+const initialState = {
+    mainLoader: false,
+    actionStatus: {},
+    globalAlert: {},
+    localAlert: {}
+}
+
+const alertReducer = (state, { error, ready, payload, meta, type }) => {
+    const actionStatus = {
+        ...state.actionStatus,
+        [type]: typeof ready === 'undefined'
+            ? undefined
+            : {
+                success: typeof ready === 'undefined' 
+                    ? typeof payload?.success === 'undefined' 
+                        ? !error
+                        : meta.alert.success
+                    : (ready ? !error : undefined),
+                pending: typeof ready === 'undefined' 
+                    ? undefined 
+                    : !ready,
+                errorType: payload?.type,
+                errorMessage: (error && payload?.toString()) || undefined,
+                data: {
+                    ...meta?.data,
+                    ...payload
+                } 
+            }
+    }
+
+    return {
+        ...state,
+        actionStatus,
+        mainLoader: typeof ready === 'undefined'
+            ? state.mainLoader
+            : Object.keys(actionStatus).reduce((x, action) => actionStatus[action]?.pending || x, false),
+        globalAlert: {
+            ...state.globalAlert,
+            [type]: (meta?.alert?.showAlert || payload?.data?.showAlert)
+                ? {
+                    show: typeof ready === 'undefined'
+                        ? true
+                        : ready && ((meta?.alert?.onlyError && error) || (meta?.alert?.onlySuccess && !error)),
+                    messageCodeHeader: meta?.alert?.messageCodeHeader || undefined,
+                    messageCode: 
+                        payload?.messageCode 
+                        || (error
+                            ? payload.type
+                                ? `reduxActions.${payload.type}`
+                                : `reduxActions.${type}.error`
+                            : `reduxActions.${type}.success`),
+                    console: error && (meta.alert?.console || payload.data?.console)
+                }
+                : undefined
+        },
+        localAlert: typeof ready === 'undefined'
```

Comment:
why there are differences in logic between `localAlert` and `globalAlert`? Like different code for `show` and `messageCode`, etc

seems like stuff would be simpler if data structure is always the same

Reply:
- `show` - for `globalAlert` it depends also on `onlyError/onlySuccess` param - at this moment we don't use these params for `localAlert` in the project
 - `messageCode` - for `globalAlert` we use `messageCode` from `WalletError` and we also use `type` from RPC error types (for example `RetriesExceeded`) - we don't use it for `localAlert`.

But I guess in both cases we can implement it, maybe in the future we can make use of these. Let me add it.